File: frontend/src/interfaces/Curriculum.ts
export interface CurriculumInterface {
    CurriculumID?: string;       // PK (string id)
    CurriculumName?: string;
    TotalCredit?: number;
    StartYear?: number;
    FacultyID?: string;
    MajorID?: string;
    BookID?: number;             // FK → BookPath.ID (int)
    Description?: string;

    // --- optional preload ---
    FacultyName?: string;
    MajorName?: string;
    BookPath?: string;           // Path ของไฟล์จริง (เช่น /static/curriculums/xxx.pdf)
}

File: backend/entity/Curriculum.go
package entity

import (
	"time"
	"gorm.io/gorm"
)

type Curriculum struct {
	CurriculumID   string `gorm:"primaryKey" json:"CurriculumID"`
	CurriculumName string `json:"CurriculumName"`
	TotalCredit    int    `json:"TotalCredit"`
	StartYear      int    `json:"StartYear"`
	Description    string `json:"Description"`

	FacultyID string   `json:"FacultyID"`
	Faculty   *Faculty `gorm:"foreignKey:FacultyID;references:FacultyID"`

	MajorID string  `json:"MajorID"`
	Major   *Majors `gorm:"foreignKey:MajorID;references:MajorID"`

	BookID int       `json:"BookID"`
	Book   *BookPath `gorm:"foreignKey:BookID;references:ID"`

	CreatedAt time.Time `json:"CreatedAt"`
	UpdatedAt time.Time `json:"UpdatedAt"`
	DeletedAt gorm.DeletedAt `json:"DeletedAt,omitempty" gorm:"index"` // ใช้สำหรับ soft delete

	Students []Students `gorm:"foreignKey:CurriculumID" json:"-"` // ระบุความสัมพันธ์ 1--many [Student]
}
